src/graph/planner/match/MatchPathPlanner.cpp

```diff
@@ -97,7 +97,11 @@ StatusOr<SubPlan> MatchPathPlanner::transform(
                             subplan,
                             nodeAliasesSeenInPattern));
 
-  MatchSolver::buildProjectColumns(qctx, path, subplan);
+  // No need to actually build path if the path is just a predicate
+  if (!path.isPred) {
```

Comment:
Better to do it in optimize rule.

Reply:
Generally ACK.
But IMHO, pattern expression and pattern predicate are semantically different, so it is also reasonable to generate different execution plans in the planner.

Reply:
Not, in general, it's properties prune problem. e.g. `where (v)-[]->()` means `List<Path>` is not empty.

Reply:
Planner should generate the most generic plan, and keep simple. Too many branches are evil.

Reply:
This pr does trigger some optimization related to property pruning, but does it have anything to do with pattern predicate. I don't get you about "in general, it's properties prune problem".

Reply:
> Planner should generate the most generic plan, and keep simple. Too many branches are evil.

Just one more if-block. Is it necessary to implement it in the optimizer phase and add a few hundred lines of code?

Reply:
I also approve of doing that in optimizer, but I still stand by my opinion.
It is true that planner should generate the most general plan, but it does not mean that a designed operator should never be generated in planner. For example, if the pattern(like (v)-[e]-(v)) is already a ExpandInto pattern, is it necessary to convert ExpandAll into ExpandInto in optimizer phase when all information is already obvious in the AST.

Reply:
ok